client/basic-authorship/src/basic_authorship.rs

```diff
@@ -210,7 +210,18 @@ impl<A, B, Block, C> ProposerInner<B, Block, C, A>
 		let mut is_first = true;
 		let mut skipped = 0;
 		let mut unqueue_invalid = Vec::new();
-		let pending_iterator = self.transaction_pool.ready();
+		let pending_iterator = match executor::block_on(future::select(
+			self.transaction_pool.ready_at(Some(self.parent_number)),
+			futures_timer::Delay::new(deadline - (self.now)()),
+		)) {
+			Either::Left((iterator, _)) => iterator,
+			Either::Right(_) => {
+				debug!(
+					"Consensus deadline reached while waiting transaction pool to reutnr pending set. Returning error!",
```

Comment:
Wouldn't it be better to at least produce an empty block instead?
```suggestion
					"Consensus deadline reached while waiting for transaction pool to return the pending set. Returning error!",
```